lib/samples/edit_with_branch_versioning/edit_with_branch_versioning.dart

```diff
@@ -358,43 +358,47 @@ class _EditWithBranchVersioningState extends State<EditWithBranchVersioning> wit
                                       await model.createVersion(parameters);
                                       setState(() {
                                         model.isVersionCreated = true;
+                                        Navigator.of(context).pop();
                                       });
-                                      Navigator.of(context).pop();
                                     } on ArcGISException catch (e) {
-                                      Navigator.of(context).pop();
-                                      // Show an error message if an exception occurs.
-                                      await showDialog<void>(
-                                        context: context,
-                                        builder: (context) {
-                                          var errorMessageStrings = e
-                                              .additionalMessage
-                                              .split(RegExp(r'\s+'));
-
-                                          if (errorMessageStrings
-                                              .contains('Extended')) {
-                                            errorMessageStrings =
-                                                errorMessageStrings.sublist(
-                                              0,
-                                              errorMessageStrings.length - 4,
-                                            );
-                                          }
-                                          final cleanedMessage =
-                                              errorMessageStrings.join(' ');
-                                          return AlertDialog(
-                                            title: const Text('Error'),
-                                            content: Text(
-                                              'Error: $cleanedMessage',
-                                            ),
-                                            actions: [
-                                              TextButton(
-                                                onPressed: () =>
-                                                    Navigator.of(context).pop(),
-                                                child: const Text('OK'),
+                                      setState(() async {
```

Comment:
Why was `setState()` added here? It seems odd that we would wrap a `showDialog` in a `setState()`.

Reply:
The reason I put the block into `setState` is to avoid the lint warning issue:
```
Don't use 'BuildContext's across async gaps, guarded by an unrelated 'mounted' check.
Guard a 'State.context' use with a 'mounted' check on the State, and other BuildContext use with a 'mounted' check on the BuildContext.dart[use_build_context_synchronously](https://dart.dev/diagnostics/use_build_context_synchronously)
```

Even I put a `if (mounted)` check, the warning still shows up.   what do you think to get rid of the warnings? or I leave the warnings as is for this PR.

Reply:
![warnings](https://github.com/user-attachments/assets/7d018c2f-ce3f-4385-b120-97e50686512e)

Reply:
I think I figured it out. Instead of `if (mounted)`, you have to do `if (context.mounted)`. This is because we're using a `StatefulBuilder` widget, which passes in a `context` parameter. So we have to check the `mounted` value on that `context` object, rather than the `mounted` value that comes from the `State` class.